src/lib/validations/ExtendedValidatorBase.ts

```diff
@@ -0,0 +1,41 @@
+import {IValidationStatus, validateBlockWithJsonSchema} from '@/lib'
+import {IBlock} from '@floip/flow-runner'
+import {ErrorObject} from 'ajv'
+import Lang from '@/lib/filters/lang'
+import {castArray} from 'lodash'
+
+const lang = new Lang()
+
+export type ValidationMessageSuffix = string
+export type ValidationResult = [ErrorObject['dataPath'], ValidationMessageSuffix]
+export type ValidationResults = ValidationResult[]
+
+export class ExtendedValidatorBase {
```

Comment:
This class encapsulates JSON schema validation and runs user-defined checks from subclasses